dist/tools/firmware_metadata/README.md

```diff
@@ -0,0 +1,36 @@
+# Metadata generator for firmware verification
+This program generates a binary file containing a metadata structure as
+follows:
+
+```c
+typedef struct firmware_metadata {
+    uint8_t hash[SHA256_DIGEST_LENGTH]; /**< SHA256 Hash of firmware image */
+    uint8_t shash[SIGN_LEN];            /**< Signed SHA256 */
+    uint16_t version;                   /**< Integer representing firmware version */
+    uint32_t size;                      /**< Size of firmware image */
+    uint32_t appid;                     /**< Integer representing the application ID */
+} firmware_metadata_t;
+```
+
+This structure is filled with the data obtained from the firmware.
+
+## Usage
+To use, you should call `generate-metadata` with the following arguments:
+
+```console
+./generate-metadata <firmware.bin> <version> <appid> <output-path>
```

Comment:
I think `<output-filename>` would lead to confusion, since we need the full path _plus_ the file name.